src/components/CreateVmWizard/CreateVmWizard.js

```diff
@@ -358,68 +348,61 @@ class CreateVmWizard extends React.Component {
                 .toJS(),
           }
         }
-
-        return update
-      })
+      }))
     }
   }
 
-  handleListOnUpdate (stepName, listName, { remove, update, create }, setStateCallback) {
-    this.setState(state => {
-      let list = this.state.steps[stepName][listName].slice(0)
+  handleListOnUpdate (stepName, listName, { remove, update, create } = {}) {
+    if (!remove && !update && !create) {
+      return
+    }
+
+    this.setState(produce(draft => {
+      const step = draft.steps[stepName]
 
       if (remove) {
-        list = list.filter(item => item.id !== remove)
+        step[listName] = step[listName].filter(item => item.id !== remove)
       }
 
       if (update) {
-        const toUpdate = list.findIndex(item => item.id === update.id)
-        if (toUpdate >= 0) {
-          list[toUpdate] = { ...list[toUpdate], ...update }
+        const toUpdate = step[listName].find(item => item.id === update.id)
+        if (toUpdate) {
+          for (const [key, val] of Object.entries(update)) {
+            toUpdate[key] = val
+          }
         }
       }
 
       if (create) {
-        list.push(create)
+        step[listName].push(create)
       }
 
-      const newState = {
-        steps: {
-          ...state.steps,
-          [stepName]: {
-            ...state.steps[stepName],
-            [listName]: list,
-            updated: (state.steps[stepName].updated + 1),
-          },
-        },
-      }
-      return newState
-    },
-    setStateCallback)
+      step.updated = step.updated + 1
+    }))
   }
 
   handleCreateVm () {
     const correlationId = generateUnique('CreateVmWizard_')
     const { basic, network: { nics }, storage: { disks } } = this.state.steps
 
-    this.setState({ correlationId })
+    this.setState(produce(draft => { draft.correlationId = correlationId }))
     this.props.onCreate(basic, nics, disks, correlationId)
   }
 
   wizardGoToStep (newStepIndex) {
-    const { activeStepIndex } = this.state
+    const { activeStepIndex, stepNavigation } = this.state
     const activeStep = this.wizardSteps[activeStepIndex]
     const newStep = this.wizardSteps[newStepIndex]
 
     // make sure we can leave the current step and enter the new step
-    if (activeStep.preventExit || newStep.preventEnter) return
+    if (stepNavigation[activeStep.id].preventExit || stepNavigation[newStep.id].preventEnter) return
 
     // run and the current step's `onExit()`
     if (activeStep.onExit) {
       activeStep.onExit()
     }
 
-    this.setState({ activeStepIndex: newStepIndex })
+    this.setState(produce(draft => { draft.activeStepIndex = newStepIndex }))
   }
```

Comment:
what about move to the previous step?
in this case prevent exit will prevent to move backward.

Reply:
Yes, that is correct and has always been that way.  Prevent exit means, "don't let the user leave this step".  Prevent enter means, "don't let the user move to this step either by direct jumping or moving forward/back".

Reply:
@sjd78 
in [1233](https://github.com/oVirt/ovirt-web-ui/pull/1233) Ben added possibility to go back regardless of `preventExit`
If this is a bad idea please comment there.